### PR TITLE
Add encrypted gpg key for code signing

### PR DESCRIPTION
**Pull request by justinharringa**

### CODE REVIEW

**Comment by justinharringa** [comment]
@afalko This should be the final step necessary to close this out unless there is some complication in decryption / import of the gpg signing key. Thanks for bearing with the churn.

**Comment by justinharringa** [comment]
The previous iteration was activating the right profile but I erroneously assumed GPG signing was being handled for us (duh... of course you need a key... argh...). The successful deploy only got the artifacts staged but javadoc & sources weren't being sent, artifacts weren't signed and Nexus also wasn't being told to auto-close.

File: .travis.settings.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
             </activation>
             <properties>
                 <gpg.executable>gpg</gpg.executable>
+                <gpg.keyname>${env.GPG_KEY_NAME}</gpg.keyname>
```
Comment (afalko) [inline .travis.settings.xml]: Can someone make a PR and have mvn echo this?

**Comment by justinharringa** [comment]
Nope. Travis hides it all. This just uses environment variable resolution.
Also the passphrase is only for signing as this is a subkey. 😄

**Comment by justinharringa** [comment]
Also, Nexus staging will validate whether this has been revoked if we find
that this has happened. Just FYI

**Pull request merged**